lib/Sema/CSStep.cpp

```diff
@@ -843,6 +843,7 @@ StepResult ConjunctionStep::resume(bool prevFailed) {
   // attempted to apply information gained from the
   // isolated constraint to the outer context.
   if (Snapshot && Snapshot->isScoped()) {
+    Snapshot.reset();
```

Comment:
I think it's a bit of an overkill just so that `)` is printed in the right spot, let's actually move `if (CS.isDebugMode())` block into `~ConjunctionStep()` instead and put it either at the bottom of the destructor or right after `Snapshot.reset()` there.